### ISSUE TITLE
Generic from super package broke completion

### ISSUE THREAD

**Issue by eliasku** [TiVo/intellij-haxe#402]
I cannot see completions and symbols are unresolved when:
Generic `a.b.B<T>` type used in inner packages `a.b.c.C`
For non-generic types all works fine.

Repro:
> Client.hx:

```
package intellij.haxe.issues.generic1.clients;
// IMPORTANT: there is no import - haxe allow to see outer package classes
class Client {
	var obj:GenericInSuperPackage<String>;
	public function new() {
		// ISSUE #1: completion doesn't work
		//obj.<caret>
		if(obj != null) {
			obj.foo(); // ISSUE #2: unresolved symbol
		}
	}
}
```
> GenericInSuperPackage.hx:

```
package intellij.haxe.issues.generic1;
class GenericInSuperPackage<T> {
	public function new() {}
	public function foo():T return null;
}
```
https://github.com/eliasku/intellij-haxe-issues/tree/master/issues/src/intellij/haxe/issues/generic1

**Comment by eliasku** [comment]
have made PR with unit tests and fix
https://github.com/TiVo/intellij-haxe/pull/403

**Comment by eliasku** [comment]
Fix https://github.com/TiVo/intellij-haxe/pull/403 merged for v0.9.9
